diff --git a/emscriptenbuild/build.sh b/emscriptenbuild/build.sh
index 2f2df5b..ba5706c 100755
--- a/emscriptenbuild/build.sh
+++ b/emscriptenbuild/build.sh
@@ -4,7 +4,8 @@ BUILD_TYPE=Debug
 ASYNCIFY_FLAGS=" -s ASYNCIFY -s 'ASYNCIFY_IMPORTS=[\"emscriptenhttp_do_get\", \"emscriptenhttp_do_read\", \"emscriptenhttp_do_post\"]' "
 POST_JS="--post-js $(pwd)/post.js"
 FS_LIBRARIES="-lidbfs.js -lnodefs.js"
-FS_EXPORTS="'FS','MEMFS','IDBFS','NODEFS','callMain','HEAPU8'"
+FS_EXPORTS="'FS','MEMFS','IDBFS','NODEFS','callMain','HEAPU8','HEAPU32','HEAP32','cwrap','ccall','UTF8ToString','getValue','setValue','stringToUTF8','lengthBytesUTF8','addFunction','removeFunction'"
+EXPORTED_FUNCS="['_malloc','_free','_main']"
 EXTRA_CMAKE_DEFINES=""
 
 # Reset in case we've done an '-async' build
@@ -67,5 +68,5 @@ if [[ "$1" != *"opfs"* ]]; then
     [ -f "../libgit2/examples/opfs_exports.c" ] && rm ../libgit2/examples/opfs_exports.c
 fi
 
-emcmake cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_C_FLAGS="$EXTRA_CMAKE_C_FLAGS --pre-js $(pwd)/pre.js $POST_JS -s \"EXPORTED_RUNTIME_METHODS=[$FS_EXPORTS]\" -sFORCE_FILESYSTEM -sEXPORT_ES6 -s INVOKE_RUN=0 -s ALLOW_MEMORY_GROWTH=1 -s STACK_SIZE=131072 $FS_LIBRARIES" -DREGEX_BACKEND=regcomp -DSONAME=OFF -DUSE_HTTPS=OFF -DBUILD_SHARED_LIBS=OFF -DTHREADSAFE=OFF -DUSE_SSH=OFF -DBUILD_CLAR=OFF -DBUILD_EXAMPLES=ON $EXTRA_CMAKE_DEFINES ..
+emcmake cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_C_FLAGS="$EXTRA_CMAKE_C_FLAGS --pre-js $(pwd)/pre.js $POST_JS -s \"EXPORTED_RUNTIME_METHODS=[$FS_EXPORTS]\" -s \"EXPORTED_FUNCTIONS=$EXPORTED_FUNCS\" -sFORCE_FILESYSTEM -sEXPORT_ES6 -s INVOKE_RUN=0 -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 -s STACK_SIZE=131072 $FS_LIBRARIES" -DREGEX_BACKEND=regcomp -DSONAME=OFF -DUSE_HTTPS=OFF -DBUILD_SHARED_LIBS=OFF -DTHREADSAFE=OFF -DUSE_SSH=OFF -DBUILD_CLAR=OFF -DBUILD_EXAMPLES=ON $EXTRA_CMAKE_DEFINES ..
 emmake make lg2
